File: 0939. Minimum Area Rectangle/Solution0939.py
class Solution0939:
    def minAreaRect(self, points: List[List[int]]) -> int:
        minArea = float('inf')
        seen = set()
        for x1, y1 in points:
            for x2, y2 in seen:
                if (x1,y2) in seen and (x2,y1) in seen:
                    minArea = min(minArea, abs(x1-x2)*abs(y1-y2))
            seen.add((x1,y1))
        return minArea if minArea != float('inf') else 0

File: 0939. Minimum Area Rectangle/Solution0939.cpp
class Solution {
public:
    int minAreaRect(vector<vector<int>>& points) {
        set<pair<int, int>> pts;
        for (vector<int> point : points) {
            pts.insert({point[0], point[1]});
        }
        int res = INT_MAX;
        for (int i = 0; i < points.size(); i++) {
            for (int j = i+1; j < points.size(); j++) {
                if (points[i][0] == points[j][0] || points[i][1] == points[j][1]) {
                    continue;
                }
                if (pts.count({points[i][0], points[j][1]}) && pts.count({points[j][0], points[i][1]}) ) {
                    res = min(res, abs(points[i][0] - points[j][0]) * abs(points[i][1] - points[j][1]));
                }
            }
        }
        return res == INT_MAX ? 0 : res;
    }
};